build.assets/tooling/cmd/build-apt-repos/apt_repo_tool.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"io/fs"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type AptRepoTool struct {
+	config       *Config
+	aptly        *Aptly
+	s3Manager    *s3manager
+	supportedOSs *map[string][]string
+}
+
+// Instantiates a new apt repo tool instance and performs any required setup/config.
+func NewAptRepoTool(config *Config, supportedOSs *map[string][]string) (*AptRepoTool, error) {
+	art := &AptRepoTool{
+		config:       config,
+		s3Manager:    NewS3Manager(*config.bucketName),
+		supportedOSs: supportedOSs,
+	}
+
+	aptly, err := NewAptly()
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to create a new aptly instance")
+	}
+	art.aptly = aptly
+
+	return art, nil
+}
+
+// Runs the tool, performing all of the business logic outside of argument parsing.
+func (art *AptRepoTool) Run() error {
+	start := time.Now()
+	logrus.Infoln("Starting APT repo build process...")
+
+	err := art.s3Manager.DownloadExistingRepo(*art.config.localBucketPath)
```

Comment:
Nit: `SyncExistingRepo` would probably be a bit more suitable name for the method.

Reply:
I think I initially used 'sync' instead of 'download', but I ended up with 'download' because 'sync' is ambiguous between uploading and downloading. For example, if I renamed `DownloadExistingRepo` to `SyncExistingRepo`, the name could more readily imply that the function makes the bucket contents match the disk contents as apposed to the other way around. Does that make sense?